src/server/routes.ts

```diff
@@ -13,4 +13,19 @@ router.get("/users", async (_req, res) => {
   res.json(users);
 });
 
+// User login
+router.post("/users/login", async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+    const user = await readUser({ email, password });
+    if (!user) {
+      res.status(404).send("Email or password incorrect");
+      return;
+    }
+    res.status(200).json(user);
```

Comment:
There is only one thing left, to actually login the user.
Your server can add a cookie to the response headers to identify the user in upcoming requests.

Reply:
Good point, I'll try that out.

Reply:
@lmachens I implemented a cookie parser. I'd be glad if you could have a second look at my code! Many thanks! 🙏